common/types/chat_saved_object_attributes.ts

```diff
@@ -88,3 +88,31 @@ export interface SendResponse {
   messages: IMessage[];
   interactions: Interaction[];
 }
+
+export type StreamChunk =
+  | {
+      event: 'patch';
+      data: Pick<SendResponse, 'interactions' | 'messages'>;
+    }
+  | {
+      event: 'error';
+      data: string;
+    }
+  | {
+      event: 'metadata';
+      data: Partial<SendResponse>;
+    }
+  | {
+      event: 'appendMessage';
+      data: {
+        messageId: string;
+        content: string;
+      };
+    }
+  | {
+      event: 'updateOutputMessage';
```

Comment:
Minor: would the name of this even slightly confusing? As this event can actually update not only message content but all other values in `IOutput` so maybe something like `updateChunk` would be more appropriate?

Reply:
IMessage is a union type of IInput and IOutput, OutputMessage means message whose type is output actually. And this action is used for updating metadata of outputMessage.

`updateChunk` to me seems to be able to describe any type of chunk from backend.